addons/controls/README.md

```diff
@@ -0,0 +1,230 @@
+
+If you're installing it into an existing project that uses `addon-knobs`, you might also [migrate your existing stories](#migrating-from-knobs).
+
+## Installation
+
+In the first version, Controls requires that you're using [Storybook Docs](https://github.com/storybookjs/storybook/tree/master/addons/docs). If you're not using it already, please install that first.
+
+Next, install the package:
+
+```sh
+npm install @storybook/addon-controls -D # or yarn
+```
+
+And add it to your `.storybook/main.js` config:
+
+```js
+module.exports = {
+  addons: ['@storybook/addon-controls'],
+};
+```
+
+Once the addon is installed you should see a `Controls` tab in the addons panel:
+
+## Writing stories
+
+Controls are only available for stories that make use of [Storybook Args](https://github.com/storybookjs/storybook/blob/next/docs/src/pages/formats/component-story-format/index.md#args-story-inputs), so if your story functions don't accept an `Args` object as the first argument, you'll need to upgrade your stories.
+
+Here's a `react` story that doesn't use args:
+
+```js
+export const Basic = () => <Button label="hello" />;
+```
+
+And here's one that does:
+
+```js
+export const Basic = (args = { label: 'hello' }) => <Button {...args} />;
+```
+
+There are a few different ways to get from here to there:
+
+- The quickest way to do this is to use [auto-generated args](#auto-generated-args).
+- If you're implementing custom story logic based on control value, you should use [manually-configured args](#manually-configured-args)
+- And if you're adding this to an existing project that's already using knobs, you should [migrate it](#migrating-from-knobs).
+
+### Auto-generated args
+
+In the example above, [Storybook Docs](https://github.com/storybookjs/storybook/tree/next/addons/docs) automatically annotates the `Button` with `ArgTypes` metadata that allows `Controls` to automatically create rows for that component.
+
+So if the full story was:
+
+```js
+import { Button } from './Button';
+export default { title: 'Button', component: Button };
+
+export const Basic = () => <Button label="hello" />;
+```
+
+You could simply rewrite it as:
+
+```js
+export const Basic = (args = { label: 'hello' }) => <Button {...args} />;
+```
+
+### Manually-configured args
+
+Auto-generated args make sense when the controls correspond exactly to component. But there are cases where you want controls to control something about the story instead of (or in addition to) the component properties.
+
+Consider the following example that shows reflow behavior of an array of `Button` instances inside a `ButtonGroup` container:
+
+```js
+import range from 'lodash/range';
+import { Button, ButtonGroup } from '.';
+
+export const Reflow = ({ count }) => (
+  <ButtonGroup>
+    {range(count).map((i) => (
+      <Button label={`button ${i}`} />
+    ))}
+  </ButtonGroup>
+);
+```
+
+In this case, we need to specify `ArgTypes` metadata about the story to render properly:
```

Comment:
According to the (guidelines)[https://github.com/storybookjs/storybook/tree/next/docs#guidelines-for-writing-good-documentation] of Storybook, avoid the use of pronouns

Reply:
I disagree with that guideline

Reply:
Then I guess.. let's discuss about them and make them in a way that matches the preferred style so that anyone following them will be consistent with the rest of the documentation?